canonical-web-and-design/maas-ui
# fix(nav): fix history when navigating between apps

## Done

- Update the header links and legacy catch-all so that there aren't multiple history events when
navigating around the app.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Click around between the react and legacy apps and use your browser back/forward buttons to move between them.
- Take note that:
  - the browser url changes
  - the page matches the url
  - the browser history only takes one click to move back/forward

## Fixes

Fixes: canonical-web-and-design/maas-ui#1718
